docs/dev/dependencies.md

```diff
@@ -28,3 +32,54 @@ If your vendor bump touched `github.com/openshift/api`, also run `go generate ./
 This [guide](https://github.com/golang/go/wiki/Modules#how-to-use-modules) is a great source to learn more about using `go mod`.
 
 For the sake of your fellow reviewers, commit vendored code separately from any other changes.
+
+### CAPI Provider Controller Dependencies
+
+The installer uses Cluster API controllers to provision the cluster insfrastructure (more details [here](./cluster-api.md)). Each provider are vendored to `cluster-api/providers/<provider-name>` (e.g. `cluster-api/providers/aws` for AWS platform).
+
+Whenever there are changes in the CAPI provider projects that need to be available to the installer, we need to bump the provider version in the installer project. The following describes the process to do so. We will use AWS provider (CAPA) as an example, but other providers should follow the same steps.
+
+#### Bump provider `go.mod`
+
+We need to set the version of CAPI provider to the revisions with the changes in the provider `go.mod`.
+
+```bash
+# At the project root (i.e. <path>/installer)
+cd cluster-api/providers/aws/
+# For example, go get sigs.k8s.io/cluster-api-provider-aws/v2@17a09f591
+go get sigs.k8s.io/cluster-api-provider-aws/v2@<commit-hash>
```

Comment:
This could be an official release tag (which would be slightly preferred)--not necessarily a commit hash (which is common). Perhaps that is worth pointing out.